packages/crayons-core/src/components/popover/popover.tsx

```diff
@@ -183,16 +192,16 @@ export class Popover {
     if (this.trigger === 'click') {
       this.triggerRef.addEventListener(this.trigger, () => {
         if (this.isOpen) {
-          this.hide();
+          this.handlePopOverHide();
```

Comment:
Can we keep the calls to show and hide here as it is and handle the required changes in `show` and `hide` methods ?

Reply:
Makes sense, corrected it. 
But, not able to handle it in a recursive way as I don't want to introduce an additional flag to use in the breaking condition. Instead added a private method aligning with above suggestion at the same time avoiding duplication of logic.

Reply:
Thanks for accommodating changes!